integration/spark/src/main/common/java/io/openlineage/spark/agent/OpenLineageSparkListener.java

```diff
@@ -186,11 +192,12 @@ public void onJobStart(SparkListenerJobStart jobStart) {
                                     .map(ds -> ds.jobIdToActiveJob().get(jobStart.jobId()))
                                     .flatMap(ScalaConversionUtils::asJavaOptional))
                         .map(job -> getSqlExecutionId(job.properties())))
-            .map(
-                id -> {
-                  long executionId = Long.parseLong(id);
-                  return getExecutionContext(jobStart.jobId(), executionId);
-                })
+            .map(id -> Long.parseLong(id))
+            .filter(
+                id ->
+                    sparkSqlExecutionRegistry.containsKey(id)
+                        || SQLExecution.getQueryExecution(id) != null)
+            .map(id -> getExecutionContext(jobStart.jobId(), id))
```

Comment:
I'd remove the `filter` here, for the same reason as above - the map is changed in a different thread. Checking before getting will likely lead to bugs. Just call `getExecutionContext`, which will first check the map, then invoke the `ContextFactory` to try to create the `ExecutionContext`. We should update the `ContextFactory` method to actually verify the `QueryExecution` we get isn't null. Here, we can filter _after_ the map call to ensure `Objects::nonNull`.
I.e., 
```java
.map(Long::parseLong)
.map(id -> getExecutionContext(jobStart.jobId(), id))
.filter(Objects::nonNull)
.orElseGet(() -> getExecutionContext(jobSTart.jobId());

Reply:
Rewriting this to `filter(Objects::nonNull)` makes the delta test fail with NullPointerExceptions, the same as in the comment above.